core/src/main/java/org/springframework/security/saml/SAMLProcessingFilter.java

```diff
@@ -89,6 +89,10 @@ public Authentication attemptAuthentication(HttpServletRequest request, HttpServ
             context.setLocalEntityEndpoint(SAMLUtil.getEndpoint(context.getLocalEntityRoleMetadata().getEndpoints(), context.getInboundSAMLBinding(), context.getInboundMessageTransport(), uriComparator));
 
             SAMLAuthenticationToken token = new SAMLAuthenticationToken(context);
+
+			// Allow subclasses to set the "details" property
+			setDetails(request, token);
```

Comment:
In lieu of calling a new method here, I'd recommend it simply call `this.token.setDetails(details)` directly.